src/lib/googleGA.ts

```diff
@@ -1,9 +1,30 @@
 import ReactGA from 'react-ga'
 export { default as ReactGA } from 'react-ga'
+export { default as TagManager } from 'react-gtm-module'
+
+export interface GUAnalytics {
+  /** Will default to 'Button Click' on `<GUButton/>` and 'Link Click' on `<Link/>` */
+  category?: string
+  action: string
+  label?: string
+  payload?: unknown
+}
 
 const randomId = (length = 10) =>
   [...Array(length)].map(() => ((Math.random() * 36) | 0).toString(36)).join('')
 
+let _dataLayer: string | undefined = undefined
+/** Returns the current data layer from Google Tag Manager */
+export const dataLayer = (): string | undefined => _dataLayer
+/**
+ * NOT to be used outside `common-front`. Used internally withing the
+ * package to avoid unnecessary hooks/complicated containers to store this
+ * the data layer used by GTM.
+ */
+export const setDataLayer = (s: string | undefined): void => {
+  _dataLayer = s
+}
+
```

Comment:
I'm still thinking whether or not to keep this approach, by defining this with a unstated container we'd have a cleaner implementation, but `guLibConfig` would only be callable inside the container (we usually call it at `const App = () => ...`, which is not wrapped in the container).

Unlike ReactGA, the `dataLayer` name is not saved globally on the default export of `react-gtm-module`, which is why I needed to do this.